documentation/src/main/asciidoc/topics/ref_config_declarative.adoc

```diff
@@ -0,0 +1,38 @@
+[id='declarative-{context}']
+= Declarative Configuration
+You configure {brandname} caches by defining properties in `infinispan.xml`.
+
+The following example shows the basic structure of a {brandname} configuration:
+
+[source,xml,options="nowrap",subs=attributes+]
+----
+<infinispan> <1>
+   <cache-container default-cache="local"> <2>
+      <transport stack="udp" cluster="mycluster"/> <3>
+      <local-cache name="local"/> <4>
+      <invalidation-cache name="invalidation"/> <5>
+      <replicated-cache name="replicated"/> <6>
+      <distributed-cache name="distributed"/> <7>
+   </cache-container>
+</infinispan>
+----
+
+<1> adds the root element for the {brandname} configuration. The minimum valid configuration is `<infinispan />`; however this provides very basic capabilities with no clustering and no cache instances.
+<2> defines properties for all caches within the container and names the default cache.
+<3> defines transport properties for clustered cache modes. In the preceding example, `stack="udp"` specifies the default JGroups UDP transport stack and names the {brandname} cluster.
+<4> names a local cache instance.
+<5> names a cache instance that uses invalidation mode.
+<6> names a cache instance that uses replicated mode.
+<7> names a cache instance that uses distributed mode.
+
+.Default caches
+{brandname} does not provide an implicit default cache. You must specify which
+named cache is the default.
+
+Additional named caches inherit settings from the default cache. You can,
+however, override or extend configuration in each named cache.
```

Comment:
I think this needs clarifying to make it clear that "additional named caches" refers to when a user calls getCache("cache-name"). It does not mean that another declaritevely named cache will inherit it's configuration from the default cache.

This is made somewhat more confusing by the fact we also have to explain <cache-type>-configuration and how multiple caches can use it via:

<distributed-cache name="repl" configuration="configuration-name"/>

Reply:
@tristantarrant I can't keep up, do calls to `getCache("new name")` create a new cache using the default config now?

Reply:
@ryanemerson thanks for clarifying that detail. tbh I wasn't sure about it and also felt it wasn't clear. I'm going to remove that sentence as it doesn't belong in the declarative config content.

Reply:
Caches no longer implicitly inherit from the default cache.

Reply:
The default cache is just any named cache that you obtain when you invoke `getCache()`